lib/core/src/Cardano/Wallet.hs

```diff
@@ -1838,7 +1842,11 @@ listTransactions ctx wid mMinWithdrawal mStart mEnd order = db & \DBLayer{..} ->
             let err = ErrStartTimeLaterThanEndTime start end
             throwE (ErrListTransactionsStartTimeLaterThanEndTime err)
         _ -> do
-            liftIO $ ti $ slotRangeFromTimeRange $ Range mStart mEnd
+            liftIO (try $ ti $ slotRangeFromTimeRange $ Range mStart mEnd) >>=
+                \case
+                    Left e@(PastHorizon{}) -> throwE (ErrListTransactionsPastHorizonException e)
```

Comment:
This is exhaustive https://input-output-hk.github.io/ouroboros-network/ouroboros-consensus/Ouroboros-Consensus-HardFork-History-Summary.html#t:PastHorizonException